ols/src/llms/providers/openai.py

```diff
@@ -0,0 +1,48 @@
+"""OpenAI provider implementation."""
+
+import logging
+
+from langchain.llms.base import LLM
+from langchain_openai import ChatOpenAI
+
+from ols import constants
+from ols.src.llms.providers import LLMProvider, register_llm_provider_as
+
+logger = logging.getLogger(__name__)
+
+
+@register_llm_provider_as(constants.PROVIDER_OPENAI)
+class OpenAI(LLMProvider):
```

Comment:
The main purpose of having the `LLMProvider` class now is to process the params. Currently, we don't know if and how we want to do that, so the provider definition can be a simple function too.